content/en/platform/corda/5.0/developing-applications/getting-started/utxo-advanced-ledger-extensions/advanced-design.md

```diff
@@ -0,0 +1,413 @@
+* On fungible state(s) updating, the sum of the consumed states that are fungible with each other must be equal to the sum of the created states that are fungible with each other.
+
+```kotlin
+public final class Update extends FungibleContractUpdateCommand<ExampleFungibleState> {
+  
+   @NotNull
+   
+    public Class<ExampleFungibleState> getContractStateType() {
+    return ExampleFungibleState.class;
+  }
+  
+   @Override
+  
+    protected void onVerify(@NotNull final UtxoLedgerTransaction transaction) {
+    
+    // Verify additional Update constraints
+    
+  }
+}
+```
+
+The `FungibleContractDeleteCommand` supports deleting existing fungible states and will verify the following constraints:
+
+* On fungible state(s) deleting, at least one fungible state input must be consumed.
+* On fungible state(s) deleting, the sum of the unscaled values of the consumed states must be greater than the sum of the unscaled values of the created states.
+* On fungible state(s) deleting, the sum of consumed states that are fungible with each other must be greater than the sum of the created states that are fungible with each other.
+
+```kotlin
+public final class Delete extends FungibleContractDeleteCommand<ExampleFungibleState> {
+  
+   @NotNull
+   
+    public Class<ExampleFungibleState> getContractStateType() {
+    return ExampleFungibleState.class;
+  }
+  
+   @Override
+  
+    protected void onVerify(@NotNull final UtxoLedgerTransaction transaction) {
+    
+    // Verify additional Delete constraints
+    
+  }
+}
+```
+
+## Designing a Fungible Contract
+
+A fungible contract can be implemented by extending the `FungibleContract` class, for example:
+
+```kotlin
+
+   public final class ExampleFungibleContract extends FungibleContract {
+  
+   @Override
+   
+   public List<Class<? extends FungibleContractCommand<?>>> getPermittedCommandTypes() {
+   return List.of(Create.class, Update.class, Delete.class);
+   
+  }
+}
+```
+
+## Identifiable API
+
+Module: identifiable
+
+Package: com.r3.corda.ledger.utxo.identifiable
+
+The identifiable API provides the component model for designing identifiable states and contracts. Identifiable states represent states that have a unique identifier that is guaranteed unique at the network level. Identifiable states are designed to evolve over time, where unique identifiers can be used to resolve the history of the identifiable state.
+
+## Designing an Identifiable State
+
+An identifiable state can be implemented by implementing the `IdentifiableState` interface, for example:
+
+```kotlin
+public final class ExampleIdentifiableState extends IdentifiableState {
+  
+   @Nullable
+   
+   private final StateRef id;
+  
+   public ExampleIdentifiableState(@Nullable final StateRef id) {
+    this.id = id;
+  }
+  
+   @Nullable
+  
+   public StateRef getId() {
+    return id;
+  }
+  
+   @NotNull
+   
+    public List<PublicKey> getParticipants() {
+    return List.of(...);
+  }
+}
+```
```

Comment:
```suggestion
```kotlin
public final class ExampleIdentifiableState extends IdentifiableState {
  @Nullable
  private final StateRef id;
  
  public ExampleIdentifiableState(@Nullable final StateRef id) {
    this.id = id;
  }
  
  @Nullable
  public StateRef getId() {
    return id;
  }
  
  @NotNull
  public List<PublicKey> getParticipants() {
    return List.of(...);
  }
}
```
```